File: bitget-node-sdk-api/src/lib/v3/UtaAccountApi.ts
import {BaseApi} from '../BaseApi';

// Covers account management and wallet operations.
// Merged from: UtaAccountController, UtaWalletController
export class UtaAccountApi extends BaseApi {

    // -------- account info --------

    // GET /api/v3/account/info
    info() {
        const url = '/api/v3/account/info';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/account/assets
    assets() {
        const url = '/api/v3/account/assets';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/account/sub-unified-assets
    subUnifiedAssets(qsOrBody: object) {
        const url = '/api/v3/account/sub-unified-assets';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/account/set-leverage
    setLeverage(qsOrBody: object) {
        const url = '/api/v3/account/set-leverage';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/account/set-hold-mode
    setHoldMode(qsOrBody: object) {
        const url = '/api/v3/account/set-hold-mode';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/account/transfer
    transfer(qsOrBody: object) {
        const url = '/api/v3/account/transfer';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/account/max-open-available
    maxOpenAvailable(qsOrBody: object) {
        const url = '/api/v3/account/max-open-available';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/transferable-coins
    transferableCoins(qsOrBody: object) {
        const url = '/api/v3/account/transferable-coins';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/financial-records
    financialRecords(qsOrBody: object) {
        const url = '/api/v3/account/financial-records';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/convert-records
    convertRecords(qsOrBody: object) {
        const url = '/api/v3/account/convert-records';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/repayable-coins
    repayableCoins() {
        const url = '/api/v3/account/repayable-coins';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/account/payment-coins
    paymentCoins() {
        const url = '/api/v3/account/payment-coins';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // POST /api/v3/account/repay
    repay(qsOrBody: object) {
        const url = '/api/v3/account/repay';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/settings
    settings() {
        const url = '/api/v3/account/settings';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // POST /api/v3/account/sub-transfer
    subTransfer(qsOrBody: object) {
        const url = '/api/v3/account/sub-transfer';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/sub-transfer-record
    subTransferRecord(qsOrBody: object) {
        const url = '/api/v3/account/sub-transfer-record';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/funding-assets
    fundingAssets(qsOrBody: object) {
        const url = '/api/v3/account/funding-assets';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/account/switch-deduct
    switchDeduct(qsOrBody: object) {
        const url = '/api/v3/account/switch-deduct';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/deduct-info
    deductInfo() {
        const url = '/api/v3/account/deduct-info';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // GET /api/v3/account/fee-rate
    feeRate(qsOrBody: object) {
        const url = '/api/v3/account/fee-rate';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/account/switch
    switchAccount() {
        const url = '/api/v3/account/switch';
        const headers = this.signer('POST', url, null)
        return this.axiosInstance.post(url, null, {headers})
    }

    // GET /api/v3/account/switch-status
    switchStatus() {
        const url = '/api/v3/account/switch-status';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // POST /api/v3/account/deposit-account
    depositAccount(qsOrBody: object) {
        const url = '/api/v3/account/deposit-account';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/account/sub-master-transfer
    subMasterTransfer(qsOrBody: object) {
        const url = '/api/v3/account/sub-master-transfer';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/max-transferable
    maxTransferable(qsOrBody: object) {
        const url = '/api/v3/account/max-transferable';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/open-interest-limit
    openInterestLimit(qsOrBody: object) {
        const url = '/api/v3/account/open-interest-limit';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/account/adjust-account-mode
    adjustAccountMode(qsOrBody: object) {
        const url = '/api/v3/account/adjust-account-mode';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // POST /api/v3/account/move-positions
    movePositions(qsOrBody: object) {
        const url = '/api/v3/account/move-positions';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/move-position-history
    movePositionHistory(qsOrBody: object) {
        const url = '/api/v3/account/move-position-history';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/delta-info
    deltaInfo() {
        const url = '/api/v3/account/delta-info';
        const headers = this.signer('GET', url, null)
        return this.axiosInstance.get(url, {headers})
    }

    // -------- wallet (deposit / withdrawal) --------

    // GET /api/v3/account/deposit-address
    depositAddress(qsOrBody: object) {
        const url = '/api/v3/account/deposit-address';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/sub-deposit-address
    subDepositAddress(qsOrBody: object) {
        const url = '/api/v3/account/sub-deposit-address';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/deposit-records
    depositRecords(qsOrBody: object) {
        const url = '/api/v3/account/deposit-records';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // GET /api/v3/account/sub-deposit-records
    subDepositRecords(qsOrBody: object) {
        const url = '/api/v3/account/sub-deposit-records';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/account/withdrawal
    withdrawal(qsOrBody: object) {
        const url = '/api/v3/account/withdrawal';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/withdrawal-records
    withdrawalRecords(qsOrBody: object) {
        const url = '/api/v3/account/withdrawal-records';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    // POST /api/v3/account/cancel-withdrawal
    cancelWithdrawal(qsOrBody: object) {
        const url = '/api/v3/account/cancel-withdrawal';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    // GET /api/v3/account/withdraw-address
    withdrawAddress(qsOrBody: object) {
        const url = '/api/v3/account/withdraw-address';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v3/UtaAccountApi.java
package com.bitget.openapi.api.v3;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface UtaAccountApi {

    // -------- account info --------

    @GET("/api/v3/account/info")
    Call<ResponseResult> info();

    @GET("/api/v3/account/assets")
    Call<ResponseResult> assets();

    @GET("/api/v3/account/sub-unified-assets")
    Call<ResponseResult> subUnifiedAssets(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/account/set-leverage")
    Call<ResponseResult> setLeverage(@Body Map<String, Object> paramMap);

    @POST("/api/v3/account/set-hold-mode")
    Call<ResponseResult> setHoldMode(@Body Map<String, Object> paramMap);

    @POST("/api/v3/account/transfer")
    Call<ResponseResult> transfer(@Body Map<String, Object> paramMap);

    @POST("/api/v3/account/max-open-available")
    Call<ResponseResult> maxOpenAvailable(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/transferable-coins")
    Call<ResponseResult> transferableCoins(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/financial-records")
    Call<ResponseResult> financialRecords(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/convert-records")
    Call<ResponseResult> convertRecords(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/repayable-coins")
    Call<ResponseResult> repayableCoins();

    @GET("/api/v3/account/payment-coins")
    Call<ResponseResult> paymentCoins();

    @POST("/api/v3/account/repay")
    Call<ResponseResult> repay(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/settings")
    Call<ResponseResult> settings();

    @POST("/api/v3/account/sub-transfer")
    Call<ResponseResult> subTransfer(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/sub-transfer-record")
    Call<ResponseResult> subTransferRecord(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/funding-assets")
    Call<ResponseResult> fundingAssets(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/account/switch-deduct")
    Call<ResponseResult> switchDeduct(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/deduct-info")
    Call<ResponseResult> deductInfo();

    @GET("/api/v3/account/fee-rate")
    Call<ResponseResult> feeRate(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/account/switch")
    Call<ResponseResult> switchAccount();

    @GET("/api/v3/account/switch-status")
    Call<ResponseResult> switchStatus();

    @POST("/api/v3/account/deposit-account")
    Call<ResponseResult> depositAccount(@Body Map<String, Object> paramMap);

    @POST("/api/v3/account/sub-master-transfer")
    Call<ResponseResult> subMasterTransfer(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/max-transferable")
    Call<ResponseResult> maxTransferable(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/open-interest-limit")
    Call<ResponseResult> openInterestLimit(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/account/adjust-account-mode")
    Call<ResponseResult> adjustAccountMode(@Body Map<String, Object> paramMap);

    @POST("/api/v3/account/move-positions")
    Call<ResponseResult> movePositions(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/move-position-history")
    Call<ResponseResult> movePositionHistory(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/delta-info")
    Call<ResponseResult> deltaInfo();

    // -------- wallet (deposit/withdrawal) --------

    @GET("/api/v3/account/deposit-address")
    Call<ResponseResult> depositAddress(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/sub-deposit-address")
    Call<ResponseResult> subDepositAddress(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/deposit-records")
    Call<ResponseResult> depositRecords(@QueryMap Map<String, String> paramMap);

    @GET("/api/v3/account/sub-deposit-records")
    Call<ResponseResult> subDepositRecords(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/account/withdrawal")
    Call<ResponseResult> withdrawal(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/withdrawal-records")
    Call<ResponseResult> withdrawalRecords(@QueryMap Map<String, String> paramMap);

    @POST("/api/v3/account/cancel-withdrawal")
    Call<ResponseResult> cancelWithdrawal(@Body Map<String, Object> paramMap);

    @GET("/api/v3/account/withdraw-address")
    Call<ResponseResult> withdrawAddress(@QueryMap Map<String, String> paramMap);
}
